corehq/apps/locations/models.py

```diff
@@ -549,6 +550,43 @@ def _remove_user(self):
             user.active = False
             user.save()
 
+    def deprecate_by(self, other_location, operation, archive=True):
+        """
+        add metadata on locations
+        for this/self location
+        1. deprecated_to: this can be a single location or a list like in case of a split
+        2. deprecation_operation: this location was deprecated by a merge/split/extract/move operation
+        3. deprecated_at: In case of split, since multiple locations will deprecate it, this would be a dict
+        with new location id mapped to the timestamp of deprecation performed
+        for other location
+        1. deprecates: note this location id on it. This can hold multiple values like in case of a merge a new
+        location would come from multiple locations
+        :param other_location: the location that deprecates this location
+        :param operation: the operation being performed, split/merge/move/extract
```

Comment:
I'd assert that `operation` is in `VALID_OPERATIONS` in the function.

Reply:
That is already checked by the validate step, if someone uploads an invalid file, i will find them.
But will also add a check before doing the operation that there are no errors from the parse.